src/inner_repl.jl

```diff
@@ -51,8 +51,9 @@ function run_repl(name2var, eval_module)
     local code_ast
     while true
         code_ast = get_user_input()
-        if haskey(actions, code_ast)
-            return code_ast # Send the codeword back
+        # TODO: use REPLMaker instead of this hack ?
+        if length(code_ast.args) >= 2 && haskey(actions, code_ast.args[2])
```

Comment:
I do not understand why this had to change.
Can you illuminate?

Reply:
Julia parser seems to wrap toplevel expressions in a `toplevel` node. For example,
```
julia> Base.parse_input_line("CC")
:($(Expr(:toplevel, :(#= none:1 =#), :CC)))
```
I could not find this in the doc but it is mentionned in [this thread](https://discourse.julialang.org/t/toplevel-expressions/6153/3) and is implemented by [this block code in the parser](https://github.com/JuliaLang/julia/blob/83d7ebaca528904e7bcfc7f032f19d44fc1acd49/src/julia-parser.scm#L717).

Reply:
Perhaps clearer then:
```suggestion
        if Meta.isexpr(code_ast, :toplevel, 2) && haskey(actions, code_ast.args[2])
```